### DISCUSSION TITLE
2.0: Refactor Discussion

### DISCUSSION THREAD

**Discussion by jjjermiah** [bhklab/med-imagetools#144] [Ideas]
Open discussion of 

- ideas
- bugs
- clarifications

anyone wants to see in the near future.

**Comment by katy-scott** [comment]
**Autopipeline Crawl**
- should include ROI names
- Can it be customizable? I don't need the MRI columns when I'm working with a CT dataset
- RECIST project will have new columns needed as well
- The absolute path setup needs to better explained to the user or updated
    - when trying to run Autopipeline as a function I can't pass in relative paths that include "../"  
    
**Better handling if a reference CT value is missing in a segmentation file**
- let a user input their own DICOM header tag if they know what it is
- maybe a config file option that lets you set those values for users who are more familiar with DICOM tags

**Comment by RuiyanNi** [comment]
**Handle hollow structure** 

- Please check if the latest version can handle hollow structures in RTSTRUCT when converting it to Nifti format

**SOP Instance UID in RTSTRUCT**

- double-check if giving errors for RTSTRUCT dcm tag named 'ReferencedSOPInstanceUID'

**Reply by skim2257** [reply]
@RuiyanNi can you privately forward us an image / file of a hollow structure? I'm wondering if the contour points are just an outline or if you're referring to donut-like structures

**Comment by katy-scott** [comment]
**Remove #_ from beginning of converted nifti directories**

* Makes indexing through that directory by patient ID annoying

**Comment by katy-scott** [comment]
It would be nice to have a function that does the full RTSTRUCT loading and conversion of structure set to a segmentation like what we have in READII ([`loadRTSTRUCTSITK` function](https://github.com/bhklab/readii/blob/6edeccc38429de5ac3f8f353f31337919b9e8bc0/src/readii/loaders.py#L44))

Also having another function in `StructureSet` like `to_segmentation` that returns a sitk.Image or array before converting it to the Segmentation object? This would be most helpful for testing the Segmentation class. https://github.com/bhklab/med-imagetools/blob/551664802d2ff21d9cf7ce6a04718c3e3f79ee06/src/imgtools/modules/structureset.py#L561-L574
